crates/ruff_linter/src/rules/airflow/rules/removal_in_3.rs

```diff
@@ -853,13 +857,31 @@ fn check_name(checker: &Checker, expr: &Expr, range: TextRange) {
         _ => return,
     };
 
-    checker.report_diagnostic(Diagnostic::new(
+    if is_guarded_by_try_except(expr, &replacement, semantic) {
+        return;
+    }
+
+    let mut diagnostic = Diagnostic::new(
         Airflow3Removal {
             deprecated: qualified_name.to_string(),
-            replacement,
+            replacement: replacement.clone(),
```

Comment:
I tried to move the fix a bit but did not succeeded. Should we refactor the fix here as well?

Reply:
I'm not sure I follow here. Do you mean to avoid the clone here that I suggested in your other PRs?

Reply:
If it requires too much effort then it's fine to avoid making the change.